pandas/tests/io/sas/test_sas7bdat.py

```diff
@@ -183,6 +183,17 @@ def test_date_time(datapath):
     tm.assert_frame_equal(df, df0)
 
 
+def test_compact_numerical_values(datapath):
+    # Regression test for #21616
+    fname = datapath("io", "sas", "data", "cars.sas7bdat")
+    df = pd.read_sas(fname, encoding='latin-1')
+    # The two columns CYL and WGT in cars.sas7bdat have column
+    # width < 8 and only contains integral values. Test
+    # that pandas doesn't corrupt the less significant bits.
+    tm.assert_series_equal(df['WGT'], df['WGT'].round(), check_exact=True)
```

Comment:
We aren't comparing to self. We are making sure, that floats with no decimal part in the sas7bdat-file doesn't get a decimal part because of a bug in pandas. That is exactly what this bug is about, after all. 

(See: https://stackoverflow.com/questions/49059421/pandas-fails-with-correct-data-type-while-reading-a-sas-file)

Before:
1. Memory was not initialized.
2. The decimal part of the float was taken from random uninitialized memory.
3. Test fails most of the time.

Now:
1. Memory is zeroed out.
2. The decimal part of the float which is not read (because it's implicit in the file format, that it should be zero) is therefore still zero in pandas.
3. Test succeeds.

Reply:
can you use
```
result = 
expected = 
```

to make this easier to read

why do we need to .round(), rather than just using ``check_less_precision`` ?

Reply:
I can also do:
```
msg = "Expected df['WGT'] to be full of integers"

assert df['WGT'] == df['WGT'].astype('int'), msg
tm.assert_series_equal(df['WGT'], df['WGT'].astype('int'), check_exact=True)
assert all(f.is_integer() for f in df['WGT']), msg
assert any(f - int(f) != 0 for f in df['WGT']), msg
assert all(f == int(f) for f in df['WGT']), msg
```
or any other way of saying the same thing you would prefer...